### PR TITLE
Opt out interfaces

### PR DESCRIPTION
**Pull request by stasberkov**
Implementation for Issue #5

### CODE REVIEW

File: rules/explicitOverrideRule.ts [hunk 1]
```diff
@@ -2,33 +2,35 @@ import * as Lint from 'tslint';
...
+    ts.MethodDeclaration |
+    ts.PropertyDeclaration |
+    ts.GetAccessorDeclaration |
+    ts.SetAccessorDeclaration;
 
 type OverrideKeyword =
-        ts.Decorator |
-        ts.JSDocTag;
+    ts.Decorator |
+    ts.JSDocTag;
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: Please keep the indentation unchanged.

File: rules/explicitOverrideRule.ts [hunk 2]
```diff
@@ -78,10 +83,10 @@ const OVERRIDE_DECORATOR_MATCHER = /^@[oO]verride(\(\s*\))?$/;
...
 
     constructor(
-            sourceFile: ts.SourceFile,
-            ruleName: string,
-            private readonly _options: IOptions,
-            private readonly checker: ts.TypeChecker) {
+        sourceFile: ts.SourceFile,
+        ruleName: string,
+        private readonly _options: IOptions,
+        private readonly checker: ts.TypeChecker) {
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: Same here. To indents is for arguments lists, one indent for code blocks.

File: rules/explicitOverrideRule.ts [hunk 3]
```diff
@@ -99,15 +104,15 @@ class Walker extends Lint.AbstractWalker<IOptions> {
...
-                    break;
+            case ts.SyntaxKind.Constructor:
+                this.checkConstructorDeclaration(element);
+                break;
+            case ts.SyntaxKind.MethodDeclaration:
+            case ts.SyntaxKind.PropertyDeclaration:
+            case ts.SyntaxKind.GetAccessor:
+            case ts.SyntaxKind.SetAccessor:
+                this.checkOverrideableElementDeclaration(element);
+                break;
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: This was actually a mistake, good catch 😉

File: rules/explicitOverrideRule.ts [hunk 4]
```diff
@@ -117,15 +122,15 @@ class Walker extends Lint.AbstractWalker<IOptions> {
         const foundKeyword = this.checkNodeForOverrideKeyword(node);
         if (foundKeyword !== undefined) {
             this.addFailureAtNode(foundKeyword, 'Extraneous override keyword',
-                    Lint.Replacement.deleteText(foundKeyword.getStart(), foundKeyword.getWidth()));
+                Lint.Replacement.deleteText(foundKeyword.getStart(), foundKeyword.getWidth()));
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: Keep the tab

File: rules/explicitOverrideRule.ts [hunk 5]
```diff
@@ -117,15 +122,15 @@ class Walker extends Lint.AbstractWalker<IOptions> {
...
         }
     }
 
     private checkConstructorDeclaration(node: ts.ConstructorDeclaration) {
         const foundKeyword = this.checkNodeForOverrideKeyword(node);
         if (foundKeyword !== undefined) {
             this.addFailureAtNode(foundKeyword, 'Extraneous override keyword: constructors always override the parent',
-                    Lint.Replacement.deleteText(foundKeyword.getStart(), foundKeyword.getWidth()));
+                Lint.Replacement.deleteText(foundKeyword.getStart(), foundKeyword.getWidth()));
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: Keep the tab

File: rules/explicitOverrideRule.ts [hunk 6]
```diff
@@ -135,7 +140,7 @@ class Walker extends Lint.AbstractWalker<IOptions> {
         if (isStaticMember(node)) {
             if (foundKeyword !== undefined) {
                 this.addFailureAtNode(foundKeyword, 'Extraneous override keyword: static members cannot override',
-                        Lint.Replacement.deleteText(foundKeyword.getStart(), foundKeyword.getWidth()));
+                    Lint.Replacement.deleteText(foundKeyword.getStart(), foundKeyword.getWidth()));
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: Keep the tab

File: rules/explicitOverrideRule.ts [hunk 7]
```diff
@@ -148,18 +153,18 @@ class Walker extends Lint.AbstractWalker<IOptions> {
 
         if (foundKeyword !== undefined && base === undefined) {
             this.addFailureAtNode(node.name, 'Member with @override keyword does not override any base class member',
-            Lint.Replacement.deleteText(foundKeyword.getStart(), foundKeyword.getWidth()));
+                Lint.Replacement.deleteText(foundKeyword.getStart(), foundKeyword.getWidth()));
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: Add a second tab

File: rules/explicitOverrideRule.ts [hunk 8]
```diff
@@ -264,7 +269,7 @@ class Walker extends Lint.AbstractWalker<IOptions> {
         }
         if (found) {
             this.addFailureAtNode(dec, `@override decorator already specified`,
-                    Lint.Replacement.deleteFromTo(dec.pos, dec.end));
+                Lint.Replacement.deleteFromTo(dec.pos, dec.end));
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: Keep the tab

File: rules/explicitOverrideRule.ts [hunk 9]
```diff
@@ -291,13 +296,13 @@ class Walker extends Lint.AbstractWalker<IOptions> {
         }
         if (found) {
             this.addFailureAtNode(child.tagName, `@override jsdoc tag already specified`,
-                    Lint.Replacement.deleteFromTo(child.pos, child.end));
+                Lint.Replacement.deleteFromTo(child.pos, child.end));
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: Keep the tab

File: rules/explicitOverrideRule.ts [hunk 10]
```diff
@@ -291,13 +296,13 @@ class Walker extends Lint.AbstractWalker<IOptions> {
...
         }
         return child;
     }
 
     private checkHeritageChain(declaration: ts.ClassDeclaration | ts.ClassExpression, node: OverrideableElement)
-            : ts.Type | undefined {
+        : ts.Type | undefined {
```
Comment (hmil) [inline rules/explicitOverrideRule.ts]: Keep the tab

**Review by hmil — CHANGES_REQUESTED** [review]
Thank you, looks good. There are a few issues with formatting (I am using two tabs for arguments lists, even though you caught some inconsistencies too).

Regarding testing, until now the behavior with respect to interfaces was left unspecified. You should add an interface in "spec.ts.lint" and make sure that tslint reports an error when the override keyword is missing and when it is extraneous.
Then, the test you added will be sufficient to ensure that interfaces are excluded when the `exclude-interfaces` flag is provided.

**Comment by stasberkov** [comment]
What editor do yo use? Do you use auto code reformat feature of code editor?  In VS Code this it triggered by Alt+Shift+F. These indent-releated changes were done by VS Code autoformat. I will rollback them if you want.

**Comment by stasberkov** [comment]
Please review

**Review by hmil — APPROVED** [review]

**Pull request merged**